api/views/application.py

```diff
@@ -16,6 +18,77 @@ def return_exception(e):
     }
     return jsonify(response_object)
 
+@application.route('/user/applications/upload', methods=['POST'])
+def upload(acl="public-read"):
```

Comment:
This is great! If possible, can you try to combine this upload functionality into the create_application endpoint? Uploading a file should not be a separate endpoint!